ixwebsocket/IXSocketConnect.cpp

```diff
@@ -56,27 +45,29 @@ namespace ix
         // block us for too long
         SocketConnect::configure(fd);
 
-        if (::connect(fd, address->ai_addr, address->ai_addrlen) == -1
-            && errno != EINPROGRESS && errno != 0)
+        int res = ::connect(fd, address->ai_addr, address->ai_addrlen);
+
+        if (res == -1 && !Socket::isWaitNeeded())
         {
-            errMsg = strerror(errno);
-            closeSocket(fd);
+            errMsg = strerror(Socket::getErrno());
+            Socket::closeSocket(fd);
             return -1;
         }
 
+        // Use select to check the status of the new connection
+        struct timeval timeout;
+        timeout.tv_sec = 0;
+        timeout.tv_usec = 10 * 1000; // 10ms timeout
+
         for (;;)
         {
             if (isCancellationRequested && isCancellationRequested()) // Must handle timeout as well
             {
-                closeSocket(fd);
+                Socket::closeSocket(fd);
                 errMsg = "Cancelled";
                 return -1;
             }
 
-            // Use select to check the status of the new connection
-            struct timeval timeout;
-            timeout.tv_sec = 0;
-            timeout.tv_usec = 10 * 1000; // 10ms timeout
             fd_set wfds;
```

Comment:
The timeout needs to be re-initialized everytime / we need a comment for that. This is a problem on Linux where the select implementation 'edits the timeout'

```


       On Linux, select() modifies timeout to reflect the amount of time not
       slept; most other implementations do not do this.  (POSIX.1 permits
       either behavior.)  This causes problems both when Linux code which
       reads timeout is ported to other operating systems, and when code is
       ported to Linux that reuses a struct timeval for multiple select()s
       in a loop without reinitializing it.  Consider timeout to be unde‐
       fined after select() returns.
```